httpx/_decoders.py

```diff
@@ -167,12 +167,13 @@ def __init__(self, encoding: typing.Optional[str] = None):
 
     def decode(self, data: bytes) -> str:
         if self.decoder is None:
-            self.decoder = codecs.getincrementaldecoder("utf-8")(errors="strict")
+            attempt_utf_8 = codecs.getincrementaldecoder("utf-8")(errors="strict")
             try:
-                return self.decoder.decode(data)
+                attempt_utf_8.decode(data)
             except UnicodeDecodeError:
-                self.decoder = codecs.getincrementaldecoder("cp1251")(errors="ignore")
-                return self.decoder.decode(data)
+                self.decoder = codecs.getincrementaldecoder("cp1252")(errors="replace")
+            else:
+                self.decoder = codecs.getincrementaldecoder("utf-8")(errors="replace")
```

Comment:
Why `errors="replace"` here if it passed with `errors="strict"` to get here?

Reply:
We need `strict` to raise an error if it doesn't appear to decode as UTF-8, but once we've made the decision we use `errors="replace"` for the most robust behaviour possible. So eg. if we've got a streaming response that initially appears to be UTF-8, but later has some non-UTF-8 bytes, then we're not raising a hard error on accessing `.text`.

(We'd like it to have a failure mode that is as graceful as possible.)